### DISCUSSION TITLE
Add support for scheduled maintenance windows

### DISCUSSION THREAD

**Discussion by alvarolopez** [upptime/upptime#122] [Ideas] [answered]
Although an issue can be added as an scheduled maintenance period, it would be nice to differentiate current and past events (i.e. incidences) with future planned activities. Adding a issue right now shows the current date and time, but little information about when it is planned.

I would suggest something like:
 - Add a tag (scheduled-mainteinance) to specify that this is something to happen in the future.
 - Add some metadata to the issue, including affected endpoints as well as an intervention date.
 - Render it under a separate section in the status page.

**Comment by AnandChowdhary** [comment]
This is a great idea, I was thinking of opening custom issues and maintenance would fit right in. You're right, we can just create issues right now, but it'll be nice to have a special UI for them.

**Comment by alvarolopez** [comment]
This would be extremely useful for us, as we want to warn users in advance of planned interventions.

**Comment by BBaoVanC** [comment]
Is there any news on this?

**Comment by AnandChowdhary** [comment, marked as answer]
You can now have scheduled maintenance events in Upptime. See issue https://github.com/upptime/upptime/issues/267 which is shows an example scheduled maintenance issue.

You should create a new issue in your @upptime repository and add the label `maintenance` to it. Then, you can add details in the comments. Make sure you also add an HTML comment at the beginning of the issue, which looks like this. If you expect the some endpoints will be down, you can add their slugs as a comma-separated list in `expectedDown`. Similarly, you can have `expectedDegraded` if you anticipate degraded performance.

```html
<!--
start: 2021-02-24T13:00:00.220Z
end: 2021-02-24T14:00:00.220Z
expectedDown: google, hacker-news
-->
```

Upptime won't open issues for sites that are expected to be down or degraded if maintenance is currently going on. When a scheduled maintenance is ongoing or in the future, you see it on top of the status page, like so:

<img width="845" alt="Screen Shot 2021-02-24 at 19 18 12@2x" src="https://user-images.githubusercontent.com/2841780/109009852-0fcae980-76d5-11eb-94ad-3a33dea6539e.png">

You'll also see previous scheduled maintenance events in their own section above "Past Incidents".

**Reply by BBaoVanC** [reply]
I ran the update and update templates CI jobs, but I don't have a maintenance tag (or template, unless there isn't a template). Should I create the maintenance tag?

**Reply by AnandChowdhary** [reply]
Yes, you can create the label yourself.
